ui/FilesPanel.go

```diff
@@ -90,33 +90,58 @@ func (this *filesPanel) createRefreshButton() *gtk.Button {
 
 func (this *filesPanel) createBackButton() *gtk.Button {
 	image := utils.MustImageFromFileWithSize("back.svg", this.Scaled(40), this.Scaled(40))
-	return utils.MustButton(image, func() {
-		if this.locationHistory.Length() < 1 {
-			this.UI.GoToPreviousPanel()
-		} else if this.locationHistory.IsRoot() {
-			this.locationHistory.GoBack()
-			this.doLoadFiles()
-		} else {
-			this.locationHistory.GoBack()
-			this.doLoadFiles()
-		}
-	})
+	return utils.MustButton(image, this.goBack)
 }
 
 func (this *filesPanel) doLoadFiles() {
 	utils.EmptyTheContainer(&this.listBox.Container)
-
-	if this.displayRootLocations() {
-		this.addRootLocations()
-	} else {
+	if this.isRoot() {
+		if this.refreshSD() {
+			this.addRootLocations()
+		} else {
+			this.locationHistory = utils.LocationHistory {
+				Locations: []dataModels.Location{dataModels.Local},
+			}
+		}
+	}
+	if !this.isRoot() {
```

Comment:
I like it.  I have a few requests.  When first reading this line, it looks odd.  Line 98 has `if this.isRoot()` an then line 107 has `if !this.isRoot()` and immediately, my thought was, "why is isRoot() being called twice?  Why not just use 'else'?"  Well, reading through the code, I know why, but this wasn't obvious or intuitive.   Please add a comment above line 107, something to the effect of
```
// this.isRoot() is called again because it is driven by locationHistory, which potentially is changed a few lines above.
```